simulator/func_sim/operation.h

```diff
@@ -48,13 +48,15 @@ std::string print_immediate( Imm type, T value)
     std::ostringstream oss;
     switch ( type)
     {
-    case Imm::ADDR:  oss << ", 0x" << std::hex << narrow_cast<uint16>(value) << std::dec; break;
-    case Imm::LOGIC: oss << ", 0x" << std::hex << value << std::dec; break;
-    case Imm::JUMP:  oss <<  " 0x" << std::hex << value << std::dec; break;
-    case Imm::TRAP:  oss << ", 0x" << std::hex << narrow_cast<int16>(value) << std::dec; break;
-    case Imm::ARITH: oss << ", "   << std::dec << narrow_cast<int16>(value); break;
-    case Imm::SHIFT: oss << ", "   << std::dec << value; break;
-    case Imm::NO:    break;
+    case Imm::ADDR:     oss << ", 0x" << std::hex << narrow_cast<uint16>(value) << std::dec; break;
+    case Imm::LOGIC:    oss << ", 0x" << std::hex << value << std::dec; break;
+    case Imm::JUMP:     oss <<  " 0x" << std::hex << value << std::dec; break;
+    case Imm::C_JUMP:   oss <<    " " << std::dec << value; break;
+    case Imm::TRAP:     oss << ", 0x" << std::hex << narrow_cast<int16>(value) << std::dec; break;
+    case Imm::ARITH:    oss << ", "   << std::dec << narrow_cast<int16>(value); break;
+    case Imm::C_ARITH:  oss << ", "   << std::dec << (narrow_cast<int8>(value << 2U) >> 2U); break;
```

Comment:
What is the intention for that operation? (<< 2 >> 2 ?)

Reply:
First byte of 6-bit immediate here influences the sign of the value. So I do '<< 2' in order to make this byte as the first byte of 8-bit immediate to know if the value is positive or negative.

Reply:
We have a special function for that:
https://github.com/MIPT-ILab/mipt-mips/blob/a15ee4e30036a92bc55eb6638c8ea134975538bb/simulator/infra/macro.h#L171

So you should do `sign_extension<8>( value);`